### PR TITLE
removes undefined objects from modules array, see issue #2382

### LINKED ISSUE #2382: Tag disabled returns undefined on getModules()
## I found a bug in MagicMirror

Please make sure to only submit reproducible issues. You can safely remove everything above the dividing line.
When submitting a new issue, please supply the following information:

**Platform**: VM Ubuntu 18.04 and https://github.com/Jopyth/MMM-Remote-Control/issues/242#issuecomment-751563402
Electron installed with MM version.

**Node Version**: v14.15.0

**MagicMirror Version**: v2.13.0

**Description**: MagicMirror gives identifiers and a place inside `getModules()` array to modules that are disabled with the `disabled` tag.

**Steps to Reproduce**: 
1) Install MagicMirror Manually
2) Add the `disabled: true,` tag inside any default module. Tried in news module.
3) Start MM with `npm start`
4) In Electron, open dev console using CTRL + SHIFT + I and write `MM.getModules()`

**Expected Results**: You should see every module described as a class, and those modules that are disabled should never be returned. Not even as `undefined`

**Actual Results**: 
image
You see not only above but in the numbers and identifiers that the number 5 is skipped as an identifier and as an integer in the array. 

**Configuration**:
```js
/*************** AUTO GENERATED BY REMOTE CONTROL MODULE ***************/

var config = 
{
  address: '0.0.0.0',
  electronOptions: {
    fullscreen: false
  },
  ipWhitelist: [],
  modules: [
    {
      module: 'MMM-Remote-Control',
      position: 'top_right',
      header: '',
      config: {
        showModuleApiMenu: true,
        customMenu: 'custom.json',
        secureEndpoints: false,
        classes: {
          test: {
            toggle: 'calendar'
          },
          'This is a test': {
            hide: [
              'calendar',
              'newsfeed'
            ],
            toggle: 'clock'
          },
          'this is another test': {
            show: [
              'calendar',
              'newsfeed'
            ],
            hide: [
              'clock'
            ]
          }
        }
      }
    },
    {
      module: 'updatenotification',
      position: 'top_bar',
      config: {}
    },
    {
      module: 'clock',
      position: 'top_left',
      header: '',
      config: {}
    },
    {
      module: 'calendar',
      header: 'US Holidays',
      position: 'top_left',
      config: {
        calendars: [
          {
            symbol: 'calendar-check',
            url: 'webcal://www.calendarlabs.com/ical-calendar/ics/76/US_Holidays.ics'
          }
        ]
      }
    },
    {
      module: 'compliments',
      position: 'lower_third',
      classes: 'test',
      config: {}
    },
    {
      module: 'newsfeed',
      position: 'bottom_bar',
      disabled: true, // Here it's the disabled module
      config: {
        feeds: [
          {
            title: 'New York Times',
            url: 'https://rss.nytimes.com/services/xml/rss/nyt/HomePage.xml'
          }
        ]
      }
    },
    {
      module: 'alert',
      header: '',
      config: {
        welcome_message: true
      }
    }
  ],
  useHttps: false,
  httpsPrivateKey: '',
  httpsCertificate: '',
  logLevel: [
    'INFO',
    'LOG',
    'WARN',
    'ERROR'
  ]
}

/*************** DO NOT EDIT THE LINE BELOW ***************/
if (typeof module !== 'undefined') {module.exports = config;}
```

**Additional Notes**: This could break code if the developers iterate above the `MM.getModules()` and brings the identifier from an `undefined` module.

### PR DESCRIPTION
**Pull request by khassel**
closes #2382 

Fix removes undefined objects from modules array.

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -502,7 +502,7 @@ var MM = (function () {
 			modules = [];
 			for (var m in moduleObjects) {
```
Comment (rejas) [inline js/main.js:503]: WOuldnt this even be easier to read with some es6 syntax like for-each?

Comment (sdetweil) [reply js/main.js:503]: but why, foreach requires a function, we got all that already..

Comment (khassel) [reply js/main.js:503]: I'm not good in js, so feel free to change ...

Comment (rejas) [reply js/main.js:503]: Would be this line instead of the whole for block:

			moduleObjects.forEach((module) => modules.push(module));

**Comment by sdetweil** [comment]
wasn't there someone trying to run the UI on a non-es6 platform.. older ios..

**Comment by rejas** [comment]
> wasn't there someone trying to run the UI on a non-es6 platform.. older ios..

yes, IOS 9 or something. There was a PR for reverting soem stuff but michmich already declined that.

**Comment by sdetweil** [comment]
ok..

**Pull request merged**